main.cpp

```diff
@@ -42,16 +42,59 @@ struct Point
     }
 private:
     float x{0}, y{0};
-};
+};
 
 template<typename Type>
 struct Wrapper
 {
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+    
+private:
+    Type val{0};
+
+};
+
+    //I'm expecting to see a separate class definition entirely, like you did with the Numeric<double> in project 4.
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val.toString() << ")" << std::endl;
+    }
+    
+private:
+    Point val{0.0f, 0.0f};
+};
+
+
+template<typename T>
+void variadicHelper(T&& others)
+{
+    Wrapper<T>(std::forward<T>(others)).print(); 
+ 
+}
+
+template<typename T, typename ... Args>
+void variadicHelper(T&& first, Args&& ... everythingElse)
+{
+    Wrapper<T>(std::forward<T>(first)).print();
+    variadicHelper(std::forward<Args>(everythingElse)...); //are these passed by copy or... ?
+}
```

Comment:
Project9.1 challenge 2:
There is another recursive solution.  it is possible for `Args ...` to be empty.
your task: remove your implementation for instruction `4)` and implement this alternate solution.
implement it **after** your existing `variadicHelper` function.
hint: what does the **recursive call** actually look like when called with an empty Variadic Parameter list?